projects/core/src/error-handling/error-interceptors/error-interceptor.service.spec.ts

```diff
@@ -0,0 +1,154 @@
+    interceptorsOrder.push(this.interceptorValue);
+    // do not pass the error to the next interceptor
+  }
+}
+
+describe('ErrorInterceptorService', () => {
+  let errorInterceptorService: ErrorInterceptorService;
+
+  beforeEach(() => {
+    interceptorsOrder = [];
+  });
+
+  describe('when there are no error interceptors', () => {
+    beforeEach(() => {
+      TestBed.configureTestingModule({
+        providers: [ErrorInterceptorService],
+      });
+
+      errorInterceptorService = TestBed.inject(ErrorInterceptorService);
+    });
+
+    it('should not call interceptors', () => {
+      const error = new Error('test error');
+      errorInterceptorService.interceptorsChain(error);
+      expect(interceptorsOrder).toEqual([]);
+    });
+  });
+
+  describe('when there are error interceptors', () => {
+    beforeEach(() => {
+      TestBed.configureTestingModule({
+        providers: [
+          ErrorInterceptorService,
+          {
+            provide: ERROR_INTERCEPTORS,
+            useClass: MockErrorInterceptorA,
+            multi: true,
+          },
+          {
+            provide: ERROR_INTERCEPTORS,
+            useClass: MockErrorInterceptorB,
+            multi: true,
+          },
+          {
+            provide: ERROR_INTERCEPTORS,
+            useClass: MockErrorInterceptorC,
+            multi: true,
+          },
+        ],
+      });
+
+      errorInterceptorService = TestBed.inject(ErrorInterceptorService);
+    });
+
+    it('should call all error interceptors with the correct order', () => {
+      const error = new Error('test error');
+      errorInterceptorService.interceptorsChain(error);
+      expect(interceptorsOrder).toEqual([
+        'C - high priority',
+        'A - normal priority',
+        'B - low priority',
+      ]);
+    });
+  });
+
+  describe('when interceptor in the middle breaks the chain', () => {
+    beforeEach(() => {
+      TestBed.configureTestingModule({
+        providers: [
+          ErrorInterceptorService,
+          {
+            provide: ERROR_INTERCEPTORS,
+            useClass: MockErrorInterceptorA,
+            multi: true,
+          },
+          {
+            provide: ERROR_INTERCEPTORS,
+            useClass: MockErrorInterceptorB,
+            multi: true,
+          },
+          {
+            provide: ERROR_INTERCEPTORS,
+            useClass: MockErrorInterceptorC,
+            multi: true,
+          },
+          {
+            provide: ERROR_INTERCEPTORS,
+            useClass: MockErrorInterceptorNoPass,
+            multi: true,
+          },
+        ],
+      });
+
+      errorInterceptorService = TestBed.inject(ErrorInterceptorService);
+    });
+
+    it('should break the chain if any interceptor does not pass an error to the next interceptor', () => {
+      errorInterceptorService.interceptorsChain(new Error('test error'));
+      expect(interceptorsOrder).toEqual([
+        'C - high priority',
+        'A - normal priority',
+        'No Pass',
```

Comment:
Nice

![Gandalf: You shall not pass!](https://media.giphy.com/media/njYrp176NQsHS/giphy-downsized-large.gif)